grammar.js

```diff
@@ -663,16 +657,10 @@ module.exports = grammar({
       ')'
     ),
 
-    call_expression: $ => prec(PREC.primary, choice(
-      seq(
-        field('function', alias(choice('new', 'make'), $.identifier)),
-        field('arguments', alias($.special_argument_list, $.argument_list))
-      ),
-      seq(
-        field('function', $._expression),
-        field('type_arguments', optional($.type_arguments)),
-        field('arguments', $.argument_list)
-      )
```

Comment:
The general motivation for this PR and my recent related ones was to factor the TS grammar for Go into a form that makes it easier to express the rules for converting syntax to [stack graphs](https://github.com/github/stack-graphs). The particular form of those rules, as TS query patterns that emit graph nodes/edges, is quite limited in its expressiveness, making it infeasible to emit the correct stack graph unless the syntax tree has been factored into a particular form. I've shared my concern with the team, along with my belief that an ALGOL-like notation would make it easier to develop, understand, and debug the TS->SG conversion algorithm for any language, and in particular to express whatever workarounds are needed due to differences between the TS syntax tree and an ideal compiler's tree. 

Since my time for this work is limited, I've restricted my scope to writing that algorithm for Go and leaving the TS factorings for another day. So you can ignore this and my other PRs.

Thanks for your time. I wasn't aware of Tree Sitter until this year, but it's a really neat piece of infrastructure.

Reply:
Ok, that makes sense. These definitely seem like good changes, in terms of making the syntax trees easier to semantically analyze. So if you want to pick them back up at any point, let me know.